temporal-sdk/src/main/java/io/temporal/client/WorkflowStartOperationUpdate.java

```diff
@@ -0,0 +1,258 @@
+            throw new IllegalArgumentException(
+                "Method '" + updateMethodName + "' does not have UpdateMethod annotation");
+          }
+          updateName = method.getName();
+          if (!annotation.name().isEmpty()) {
+            updateName = annotation.name();
+          }
+        }
+      }
+      if (resultClass == null) {
+        throw new IllegalArgumentException(
+            "Update method with name '" + updateMethodName + "' not found on: " + workflowClass);
+      }
+    } catch (ClassNotFoundException e) {
+      throw new IllegalArgumentException("Invalid method reference", e);
+    }
+
+    return new Builder<>(
+        UpdateOptions.<R>newBuilder().setUpdateName(updateName).setResultClass(resultClass),
+        workflowType,
+        args);
+  }
+
+  private final UpdateOptions<R> options;
+
+  private final Object[] args;
+
+  private final CompletableFuture<UpdateHandle<R>> handle;
+
+  private final String workflowType;
+
+  private WorkflowStartOperationUpdate(
+      UpdateOptions<R> options, String workflowType, Object[] args) {
+    this.options = options;
+    this.args = args;
+    this.workflowType = workflowType;
+    this.handle = new CompletableFuture<>();
+  }
+
+  public UpdateOptions<R> getOptions() {
+    return options;
+  }
+
+  public Object[] getArgs() {
+    return this.args;
+  }
+
+  public String getWorkflowType() {
+    return workflowType;
+  }
+
+  @Override
+  public boolean cancel(boolean mayInterruptIfRunning) {
+    return handle.cancel(mayInterruptIfRunning);
+  }
+
+  @Override
+  public boolean isCancelled() {
+    return handle.isCancelled();
+  }
+
+  @Override
+  public boolean isDone() {
+    return handle.isDone();
+  }
+
+  public UpdateHandle<R> get() throws ExecutionException, InterruptedException {
+    return handle.get();
+  }
+
+  public void complete(UpdateHandle<R> updateHandle) {
+    this.handle.complete(updateHandle);
+  }
+
+  @Override
+  public UpdateHandle<R> get(long timeout, TimeUnit unit)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    return handle.get(timeout, unit);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    WorkflowStartOperationUpdate<?> that = (WorkflowStartOperationUpdate<?>) o;
+    return Objects.equals(options, that.options)
+        && Objects.deepEquals(args, that.args)
+        && Objects.equals(workflowType, that.workflowType);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(options, Arrays.hashCode(args), workflowType);
+  }
+
+  @Override
+  public String toString() {
+    return "WorkflowStartOperationUpdate{"
+        + "options="
+        + options
+        + ", args="
+        + Arrays.toString(args)
+        + ", handle="
+        + handle
+        + '}';
+  }
+
+  public static final class Builder<R> {
+    private final UpdateOptions.Builder<R> options;
```

Comment:
The current builder setup prevents users from writing something like:

```java
<R> void applyDefaultUpdateOptions(UpdateOptions.Builder<R> builder) { ... }
```

That they can use for this update and other updates. This is probably not a big deal, but at least in Go they can write this helper. I just want to make sure we acknowledge that, hopefully unlike our other SDKs, there is no shared options building reuse in Java, and we'll have to make sure every future option we add to the `UpdateOptions` that applies to update with start gets an associated setter here.

Reply:
Apologies, I overlooked this one yesterday. To make sure I understand this right: the idea is to provide the above method on the Update operation builder? And it would set the operation's update options to the options of `builder.build()`?